siliconcompiler/remote/client.py

```diff
@@ -395,25 +489,27 @@ def fetch_results(chip):
 
     # Unauthenticated jobs get a gzip archive, authenticated jobs get nested archives.
     # So we need to extract and delete those.
-    with tarfile.open(f'{job_hash}.tar.gz', 'r:gz') as tar:
-        tar.extractall()
+    with tarfile.open(f'{job_hash}{node}.tar.gz', 'r:gz') as tar:
+        tar.extractall(path=(node if node else ''))
     # Remove the results archive after it is extracted.
-    os.remove(f'{job_hash}.tar.gz')
+    os.remove(f'{job_hash}{node}.tar.gz')
 
     # Remove dangling symlinks if necessary.
     for import_link in glob.iglob(job_hash + '/' + top_design + '/**/*',
                                   recursive=True):
         if os.path.islink(import_link):
             os.remove(import_link)
     # Copy the results into the local build directory, and remove the
-    # unzipped directory (including encrypted archives).
-    utils.copytree(job_hash,
+    # unzipped directory.
+    basedir = os.path.join(node, job_hash) if node else job_hash
+    utils.copytree(basedir,
```

Comment:
is any of this needed if we are archiving the node, shouldn't we just untar in place?

Reply:
The build directory gets renamed on the server-side to include the job ID, so it won't be the same as the client's `('option', 'builddir')`. Hence the renaming.